sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubClient.java

```diff
@@ -296,6 +296,18 @@ public static TopicPath topicPathFromName(String projectId, String topicName) {
     return new TopicPath(String.format("projects/%s/topics/%s", projectId, topicName));
   }
 
+  boolean topicExistsInProject(String projectId, String topicName) throws IOException {
+    if (projectId == null) {
+      throw new IllegalArgumentException("projectId must not be null");
+    }
+    if (topicName == null) {
+      throw new IllegalArgumentException("topicName must not be null");
+    }
+    boolean topicExists = listTopics(projectPathFromId(projectId))
```

Comment:
This may be expensive; I believe it would be better to use the `getTopic` API exposed by the `PubSub` publisher interface (ref. https://cloud.google.com/pubsub/docs/reference/rpc/google.pubsub.v1#google.pubsub.v1.Publisher.GetTopic)